### PR TITLE
screenshot tests

### PR DESCRIPTION
**Pull request by lojbrooks**
# 📝 Description

Added screenshot testing library and tests for 2 components as examples.

Github Issue
https://github.com/hmrc/android-components/issues/6

### CODE REVIEW

File: screenshots/src/androidTest/java/uk/gov/hmrc/components/screenshots/atom/bullet/BulletedTextViewTest.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import uk.gov.hmrc.components.screenshots.capture
+import uk.gov.hmrc.components.screenshots.rules.FontScale
+import uk.gov.hmrc.components.screenshots.rules.FontScaleRule
+import uk.gov.hmrc.components.screenshots.rules.NightMode
+import uk.gov.hmrc.components.screenshots.rules.NightModeRule
+
+class BulletedTextViewTest : ScreenshotTest {
+
+    @get:Rule
+    val darkModeRule = NightModeRule()
```
Comment (jemmaSlater) [inline screenshots/src/androidTest/java/uk/gov/hmrc/components/screenshots/atom/bullet/BulletedTextViewTest.kt:35]: For consistency, should we call this `nightModeRule` seeing as we refer to it as night mode everywhere else here?

File: screenshots/src/androidTest/java/uk/gov/hmrc/components/screenshots/atom/button/IconButtonTest.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import uk.gov.hmrc.components.screenshots.capture
+import uk.gov.hmrc.components.screenshots.rules.FontScale
+import uk.gov.hmrc.components.screenshots.rules.FontScaleRule
+import uk.gov.hmrc.components.screenshots.rules.NightMode
+import uk.gov.hmrc.components.screenshots.rules.NightModeRule
+
+class IconButtonTest : ScreenshotTest {
+
+    @get:Rule
+    val darkModeRule = NightModeRule()
```
Comment (jemmaSlater) [inline screenshots/src/androidTest/java/uk/gov/hmrc/components/screenshots/atom/button/IconButtonTest.kt:40]: same comment here: `nightModeRule` ?

**Review by jemmaSlater — COMMENTED** [review]
`screenshots/consumer-rules.pro` is an empty file. Is it needed?

File: screenshots/build.gradle [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+}
+apply from: "./screenshots.gradle"
+
+android {
+    compileSdkVersion 29
+    buildToolsVersion "30.0.1"
+
+    defaultConfig {
+        minSdkVersion 21
+        targetSdkVersion 29
```
[the remark this answers is not in the document]
Comment (lojbrooks) [reply screenshots/build.gradle]: I have changed to target the same version as the other modules. Updating them all to 30 is probably out of scope for this PR.

**Review by jemmaSlater — APPROVED** [review]

**Pull request merged**